test/test_op_factory.py

```diff
@@ -119,3 +120,40 @@ def matmul(
         actual = matmul(x, y)
         assert isinstance(actual, Tensor)
         assert_close(actual, Tensor(actual_data, inputs))
+
+
+@pytest.mark.skipif(get_backend() != "torch", reason="requires nn.Module")
+@pytest.mark.parametrize("batch_shape", [(), (10,)], ids=str)
+def test_nn_module(batch_shape):
+    import torch
+
+    class Matmul(torch.nn.Module):
+        def forward(
+            self, x, y
```

Comment:
One potential problem with converting `Module`s to `Op`s is that preexisting `Module`s may be especially likely to have type signatures that are not compatible with `make_op`. For example, someone might have written `Matmul` as
```py
def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
    ...
```
Should we do any preprocessing of the signature to either convert it to a Funsor-compatible form, raise an error, or raise a warning reminding the user to register a `find_domain` implementation for the new op?

Reply:
Hopefully things just break with `torch.Tensor` signatures, and users can wrap their modules with funsor compatible types.  That's out of scope for this little Sunday morning PR though.